homeassistant/components/fritz/common.py

```diff
@@ -0,0 +1,266 @@
+"""Support for AVM Fritz!Box classes."""
+from collections import namedtuple
+from datetime import datetime, timedelta
+import logging
+import socket
+from typing import Any, Dict, Optional
+
+# pylint: disable=import-error
+from fritzconnection import FritzConnection
+from fritzconnection.core.exceptions import FritzConnectionException
+from fritzconnection.lib.fritzhosts import FritzHosts
+from fritzconnection.lib.fritzstatus import FritzStatus
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_DEVICES,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import dt as dt_util, get_local_ip
+
+from .const import (
+    ATTR_HOST,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_USERNAME,
+    DOMAIN,
+    ERROR_CONNECTION_ERROR,
+    TRACKER_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+Device = namedtuple("Device", ["mac", "ip", "name"])
+
+
+def ensure_unique_hosts(value):
+    """Validate that all configs have a unique host."""
+    vol.Schema(vol.Unique("duplicate host entries found"))(
+        [socket.gethostbyname(entry[CONF_HOST]) for entry in value]
```

Comment:
This can block so it needs to be run in the executor

Reply:
I think you can take it out though since yaml config is going away anyways

Reply:
Removed